fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/plans/RecordQueryCoveringIndexPlan.java

```diff
@@ -173,9 +172,11 @@ public boolean hasLoadBykeys() {
     @Nonnull
     @Override
     public Value getResultValue() {
-        return new IndexedValue(indexPlan.getMatchCandidateOptional()
-                .map(matchCandidate -> matchCandidate.getTraversal().getRootReference().getResultType())
-                .orElse(new Type.Any()));
+        // TODO This should generate a value whose result type are the parts of what the index returns flattened out
```

Comment:
Could you please help me understand this better?
Say I have a record type R(A,B,C,D) with Index Idx(C,A)
If this index is chosen for the access path presented by this plan, does it mean that:
1. The result value should be IndexValue(C,A) with type R`(C,A)
2. The result value should be IndexValue(C,A) with type R(A, B(assuming null/omitted), C, D(assuming null/omitted)) 

Or is it maybe something else?

Reply:
There is some sort of unification happening between what is matched/what is inserted as the actual index scan and what comes out of the fetch (that fetch can be part of the index scan or delayed by using an explicit fetch operator downstream).

Reply:
Resolved offline.